### PR TITLE
HAAR-822: Added 'groups/child/{group}'' - child group details, endpoint

### PR DESCRIPTION
**Pull request by ravmoj**
Added 'groups/child/{group}'' - child group details, endpoint

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/externalusersapi/resource/GroupsController.kt [hunk 1]
```diff
@@ -68,7 +72,61 @@ class GroupsController(
...
+    @PathVariable
+    group: String,
+    @Parameter(
+      description = "Details of the child group to be updated.",
+      required = true
+    ) @RequestBody
+    groupAmendment: GroupAmendment
+
+  ) {
+    groupsService.updateChildGroup(authenticationFacade.currentUsername, group, groupAmendment)
```
Comment (karenmillermoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/externalusersapi/resource/GroupsController.kt:121]: We don't need to pass the username through here - the authenticationFacade can just be added to the constructor of the service.  I suppose it doesn't really matter, but this is how we did it for the RoleService.  Probably worth changing either RolesController/Service or GroupController/Service to be consistent.  It doesn't matter which.
Happy for this to be a separate PR

File: src/test/kotlin/uk/gov/justice/digital/hmpps/externalusersapi/resource/GroupsControllerIntTest.kt [hunk 2]
```diff
@@ -131,4 +133,57 @@ class GroupsControllerIntTest : IntegrationTestBase() {
...
+      webTestClient
+        .put().uri("/groups/child/CHILD_9")
+        .headers(setAuthorisation("ITAG_USER_ADM", listOf("ROLE_MAINTAIN_OAUTH_USERS")))
+        .body(BodyInserters.fromValue(mapOf("groupName" to "new group name")))
+        .exchange()
+        .expectStatus().isOk
+    }
+
+    @Test
+    fun `Change group name endpoint returns forbidden when dose not have admin role `() {
```
Comment (karenmillermoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/externalusersapi/resource/GroupsControllerIntTest.kt:150]: typo does/dose

File: src/test/kotlin/uk/gov/justice/digital/hmpps/externalusersapi/service/GroupsServiceTest.kt [hunk 3]
```diff
@@ -1,21 +1,27 @@
...
   private val groupRepository: GroupRepository = mock()
   private val maintainUserCheck: MaintainUserCheck = mock()
+  private val childGroupRepository: ChildGroupRepository = mock()
+  private val telemetryClient: TelemetryClient = mock()
   private val groupsService = GroupsService(
     groupRepository,
-    maintainUserCheck
+    maintainUserCheck,
+    childGroupRepository,
+    telemetryClient
```
Comment (karenmillermoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/externalusersapi/service/GroupsServiceTest.kt:24]: Should there be a check for the telemetryclient  being called in your test?

**Review by karenmillermoj — APPROVED** [review]
lgtm

File: src/test/kotlin/uk/gov/justice/digital/hmpps/externalusersapi/service/GroupsServiceTest.kt (hunk 3 above)
Comment (ravmoj) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/externalusersapi/service/GroupsServiceTest.kt]: Updated

File: src/main/kotlin/uk/gov/justice/digital/hmpps/externalusersapi/resource/GroupsController.kt (hunk 1 above)
Comment (ravmoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/externalusersapi/resource/GroupsController.kt]: Good point, Moved to service

**Review by karenmillermoj — APPROVED** [review]
lgtm
